### PR TITLE
🚥 allow CONTENT_CDN host to be spec'ed

### PR DESCRIPTION
**Pull request by stevejpurves**
fixing #342 

This allows the hostname used in the `CONTENT_CDN` to be specified independent of `PORT` or `HOST`. 

Note trying to use `HOST` for this purpose is problematic as the express development server attempts to bind to this and fails, we really just want to influence rewriting on urls to static assets, so we are adding `CONTENT_CDN_HOST` in a similar vein to how `CONTENT_CDN_PORT` is used.

A parallel change in `mystmd` is needed in order to add `CONTENT_CDN_HOST` based on `HOST` when starting the dev server (potentially only when `--keep-host` is used?)

### CODE REVIEW

**Comment by stevejpurves** [comment]
tested this by setting a name on `/etc/hosts` and then `HOST=http://noplacelikehome myst start --keep-host` with the output from `make build-book` copied into `_build/templates` folder

image

File: themes/article/app/utils/loaders.server.ts [hunk 1]
```diff
@@ -34,6 +38,7 @@ function updateLink(url: string) {
   if (process.env.MODE === 'static') {
     return `/myst_assets_folder${url}`;
   }
+  console.log('rewriting:', url, `${CONTENT_CDN}${url}`);
```
Comment (rowanc1) [inline themes/article/app/utils/loaders.server.ts:41]:
```suggestion
```

File: themes/book/app/utils/loaders.server.ts [hunk 2]
```diff
@@ -34,6 +38,7 @@ function updateLink(url: string) {
   if (process.env.MODE === 'static') {
     return `/myst_assets_folder${url}`;
   }
+  console.log('rewriting:', url, `${CONTENT_CDN}${url}`);
```
Comment (rowanc1) [inline themes/book/app/utils/loaders.server.ts:41]:
```suggestion
```

**Comment by rowanc1** [comment]
I think the approach in #468 is a bit cleaner. Allows to be set without a port. I think this also is a solution for #342.

**Pull request closed without merging**